cads_adaptors/adaptors/roocs/__init__.py

```diff
@@ -0,0 +1,85 @@
+import os
+from typing import BinaryIO
+
+from cads_adaptors.adaptors.cds import AbstractCdsAdaptor, Request
+
+os.environ["ROOK_URL"] = "http://rook.dkrz.de/wps"
+
+
+class RoocsCdsAdaptor(AbstractCdsAdaptor):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.facets = self.config.get("facets", dict())
+        self.facet_groups = self.config.get("facet_groups", dict())
+
+    def retrieve(self, request: Request) -> BinaryIO:
+        from cads_adaptors.tools import download_tools
+
+        workflow = self.construct_workflow(request)
+        response = workflow.orchestrate()
+
+        try:
+            urls = response.download_urls()
+        except Exception:
+            raise Exception(response.status)
+
+        return download_tools.DOWNLOAD_FORMATS["zip"](urls)
+
+    def construct_workflow(self, request):
+        import rooki.operators as rookops
+
+        from cads_adaptors.adaptors.roocs import operators
+
+        facets = self.find_facets(request)
+
+        dataset_id = ".".join(facets.values())
```

Comment:
This looks odd to me, but maybe you just need to explain. Does the order of the facet values matter for the dataset_id, they may have changed order when stored in the adaptor.json